packages/itwin/imodel-react-hooks/src/utils/basic-hooks.ts

```diff
@@ -25,19 +25,19 @@ export function useOnMountInRenderOrder(effect: () => void | (() => void)) {
 
 /** run code on component mount */
 export function useOnMount(impl: () => void) {
-  useEffect(() => void impl(), []);
+  useEffect(() => void impl(), [impl]);
```

Comment:
this actually should not take a dependency. It's intended to not run on additional effect runs, just the first one (i.e. mount). So this way the user passing a callback will not be surprised that it changes. Otherwise we would need to expose the dependencies to them.

An alternative explicit way of doing it is capturing the first `impl` value with either `useRef` or `useState` and using that as the dependency, since we know it won't change without us changing it

Reply:
reverted